RadixWallet/Core/DesignSystem/ToggleView.swift

```diff
@@ -30,10 +33,27 @@ public struct ToggleView: SwiftUI.View {
 							.padding(.trailing, .medium3)
 					}
 
-					PlainListRowCore(context: .toggle, title: title, subtitle: subtitle)
+					PlainListRowCore(context: context.plainListRowContext, title: title, subtitle: subtitle)
 				}
 			}
 		)
 		.frame(maxWidth: .infinity, minHeight: minHeight)
 	}
 }
+
+// MARK: ToggleView.Context
+extension ToggleView {
+	public enum Context {
+		case settings
+		case toggle
+	}
+}
+
+extension ToggleView.Context {
+	var plainListRowContext: PlainListRowCore.ViewState.Context {
+		switch self {
+		case .settings: .settings
+		case .toggle: .toggle
+		}
```

Comment:
why do we need this ?

`PlainListRowCore.ViewState.Context` sent to `PlainListRow` should always be `.toggle`.

Reply:
>PlainListRowCore.ViewState.Context sent to PlainListRow should always be .toggle.

For toggle views in Preferences, we need to use `.settings` since they have a different design in that context.


I explicitly defined `ToggleView.Context` (instead of directly passing `PlainListRowCore.ViewState.Context`) to make it clear which contexts are supported by `ToggleView`. This prevents the mistaken use of unsupported contexts, such as `.dappAndPersona`, under the assumption that they are valid states and would be handled.

Reply:
oh I see now, thanks for explanation